edb/edgeql/compiler/viewgen.py

```diff
@@ -77,6 +77,7 @@ def process_view(
         return view_scls
 
     with ctx.newscope(fenced=True) as scopectx:
+        scopectx.in_temp_scope = True
```

Comment:
Do we need a second `in_temp_scope` flag?  Is it always correlated with `path_scope.is_temporary`?  Perhaps test for that instead (add a `ScopeTree` helper if traversal is needed)?

Reply:
I mostly did it this way to avoid chasing up the tree but that's gonna be super marginal so I think you are right